src/registers/hpfar_el2.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// Copyright (c) 2024 by the author(s)
+//
+// Author(s):
+//   - Sangwan Kwon <sangwan.kwon@samsung.com>
+
+//! Hypervisor IPA Fault Address Register - EL2
+//!
+//! Holds the faulting IPA for some aborts on a stage 2 translation taken to EL2.
+
+use tock_registers::{interfaces::Readable, register_bitfields};
+
+register_bitfields! {u64,
+    pub HPFAR_EL2 [
+        /// Faulting IPA address space.
+        NS   OFFSET(63) NUMBITS(1) [],
+
+        /// Faulting Intermediate Physical Address.
+        FIPA OFFSET(4) NUMBITS(40) []
```

Comment:
Looking at [possible encodings](https://developer.arm.com/documentation/ddi0601/2024-09/AArch64-Registers/HPFAR-EL2--Hypervisor-IPA-Fault-Address-Register?lang=en) it is probably safe to encode FIPA as `OFFSET(0) NUMBITS(48)`.

Reply:
After taking a closer look, FIPA could be `OFFSET(4) NUMBIT(48)`. It's a bit confusing when viewed on the web due to the indentation, but it's clearer in the [PDF](https://documentation-service.arm.com/static/66faba6f7243ed1626434841?token=) version.

This conditions below refer to an offset within the 44 bits of FIPA[47:4].

- NS, bit [63]
- Bits [62:48]
- FIPA, bits [47:4]
  -  FIPA encoding when FEAT_D128 is implemented
     - FIPA, bits [43:0]: Bits [55:12] of the Faulting Intermediate Physical Address.
  - FIPA encoding when FEAT_LPA is implemented and FEAT_D128 is not implemented
    - Bits [43:40] Reserved, RES0.
    - FIPA, bits [39:0]: Bits [51:12] of the Faulting Intermediate Physical Address
  - FIPA encoding when FEAT_LPA is not implemented
    - Bits [43:36]: Reserved, RES0.
    - FIPA, bits [35:0]: Bits[47:12] of the Faulting Intermediate Physical Address.
- Bits [3:0]: Reserved, RES0.

Reply:
Ah, alright, yeah that makes sense, it was a bit unclear in the web version.